vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/experiments/experiment/ExperimentDetailsComponent.java

```diff
@@ -152,6 +161,39 @@ private void configureComponent() {
     addListenerForNewVariableEvent();
   }
 
+  private void initButtonBar() {
+    Button editButton = new Button("Edit");
+    editButton.addClickListener(event -> generateExperimentInformationDialog());
+    buttonBar.add(editButton);
+  }
+
+  private void generateExperimentInformationDialog() {
+    ExperimentId experimentId = context.experimentId().orElseThrow();
+    Optional<Experiment> experiment = experimentInformationService.find(experimentId);
+    experiment.ifPresentOrElse(exp -> {
+          var editDialog = ExperimentInformationDialog.prefilled(experimentalDesignSearchService,
+              exp.getName(), exp.getSpecies(),
+              exp.getSpecimens(), exp.getAnalytes());
+          editDialog.setConfirmButtonLabel("Save");
+          editDialog.addCancelEventListener(
+              experimentInformationDialogCancelEvent -> editDialog.close());
+          editDialog.addConfirmEventListener(experimentInformationDialogConfirmEvent -> {
+            ExperimentInformationContent experimentInformationContent = experimentInformationDialogConfirmEvent.getSource()
+                .content();
+            experimentInformationService.editExperimentInformation(experimentId,
+                experimentInformationContent.experimentName(), experimentInformationContent.species(),
+                experimentInformationContent.specimen(), experimentInformationContent.analytes());
+            editDialog.close();
+            fireEditEvent();
```

Comment:
please try to avoid multi-line lambda expressions! Try to extract meaningful methods instead

Reply:
That's a good point 👍